File: library-api/src/main/java/com/library/libraryapi/dto/book/CoverImages.java
package com.library.libraryapi.dto.book;

import com.library.libraryapi.model.Book;

public record CoverImages(
        String small,
        String thumbnail,
        String medium,
        String large
) {
    public static CoverImages from(Book book) {
        return new CoverImages(
                book.getSmallThumbnail(),
                book.getThumbnail(),
                book.getMediumImage(),
                book.getLargeImage()
        );
    }
}

File: library-flutter/lib/features/books/data/models/cover_images.dart
class CoverImages {
  final String? small;
  final String? thumbnail;
  final String? medium;
  final String? large;

  CoverImages({this.small, this.thumbnail, this.medium, this.large});

  factory CoverImages.fromJson(Map<String, dynamic> json) => CoverImages(
        small: json['small'] as String?,
        thumbnail: json['thumbnail'] as String?,
        medium: json['medium'] as String?,
        large: json['large'] as String?,
      );

  String? get best => large ?? medium ?? thumbnail ?? small;
}
